wolfssl/wolfcrypt/settings.h

```diff
@@ -326,6 +334,12 @@
     /* NOTE: cyassl_nucleus_defs.h is akin to user_settings.h */
     #include "nucleus.h"
     #include "os/networking/ssl/lite/cyassl_nucleus_defs.h"
+#elif !defined(BUILDING_WOLFSSL) && !defined(WOLFSSL_OPTIONS_H) && \
+      !defined(WOLFSSL_CUSTOM_CONFIG)
+    /* This error indicates that the settings header may not be included before
+     * other wolfSSL headers. If you are using a custom configuration method,
+     * define WOLFSSL_CUSTOM_CONFIG to override this error. */
+    #error "No configuration for wolfSSL detected, check header order"
```

Comment:
Maybe we should start off as a warning here and progress to an error. To account for some odd builds that are either still altering wolfssl/wolfcrypt/settings.h directly or trying to alter config.h.

Reply:
great idea.  done.